CromIAM/src/test/scala/cromiam/webservice/MockClients.scala

```diff
@@ -88,73 +90,73 @@ class MockCromwellClient()(implicit system: ActorSystem,
   }
 }
 
-class MockSamClient(checkSubmitWhitelist: Boolean = true,
-                    samResponseOption: Option[HttpResponse] = None)
+/**
+  * Overrides some values, but doesn't override methods.
+  */
+class BaseMockSamClient(checkSubmitWhitelist: Boolean = true)
                    (implicit system: ActorSystem,
-                      ece: ExecutionContextExecutor,
-                      materializer: ActorMaterializer)
+                    ece: ExecutionContextExecutor,
+                    materializer: ActorMaterializer)
   extends SamClient(
     "http",
     "bar",
     1,
     checkSubmitWhitelist,
     NoLogging,
     ActorRef.noSender
-  )(system, ece, materializer) {
-
-  val authorizedUserCollectionStr: String = "123456789"
-  val unauthorizedUserCollectionStr: String = "987654321"
-
-  val notWhitelistedUser: String = "ABC123"
+  )(system, ece, materializer)
 
-  val userCollectionList: List[Collection] = List(Collection("col1"), Collection("col2"))
+/**
+  * Extends the base mock client with overriden methods.
+  */
+class MockSamClient(checkSubmitWhitelist: Boolean = true)
```

Comment:
I'd make this more specific, not less (eg `UserValidatingSamClient`)

Reply:
Names are hard... I we go forward with cleaning up the names in this PR I'll change this one too.

Reply:
Another case of TOL. I'll leave it up to you and won't hold back approval if you don't